### PR TITLE
Testing Grid Table

### PR DESCRIPTION
**Pull request by adityadivami**
# TITLE

## Description
Summary of changes

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)

### CODE REVIEW

File: app/cdap/components/GridTable/__tests__/GridTable.test.tsx [hunk 1]
```diff
@@ -16,26 +16,328 @@
...
-    <Router history={history}>
-      <Switch>
-        <Route>
-          <GridTable />
-        </Route>
-      </Switch>
-    </Router>
-  );
+  jest.spyOn(rxjs, 'flatMap' as any).mockImplementation((callback: any) => {
+    callback({
```
Comment (adityadivami) [inline app/cdap/components/GridTable/__tests__/GridTable.test.tsx]: move to mock

File: app/cdap/components/GridTable/__tests__/GridTable.test.tsx [hunk 2]
```diff
@@ -16,26 +16,328 @@
...
+      ],
+    });
+  });
+  it('Should check if the component is rendered', () => {
+    jest.spyOn(MyDataPrepApi, 'getWorkspace').mockImplementation(() => {
+      return {
+        pipe: () => {
+          return {
+            subscribe: (callback) => {
+              callback({
```
Comment (adityadivami) [inline app/cdap/components/GridTable/__tests__/GridTable.test.tsx]: mock

File: app/cdap/components/GridTable/__tests__/GridTable.test.tsx [hunk 3]
```diff
@@ -16,26 +16,328 @@
```
Comment (adityadivami) [inline app/cdap/components/GridTable/__tests__/GridTable.test.tsx]: check whether copyrights are added in all files

File: app/cdap/components/GridTable/components/Breadcrumb/__tests__/Breadcrumb.test.tsx [hunk 4]
```diff
@@ -25,11 +25,18 @@ const history = createHistory({
 });
 
 describe('Test Breadcrumb Component', () => {
+  const locationMock = jest.mock('react-router-dom', () => ({
+    // ...jest.requireActual('react-router-dom'),
```
Comment (adityadivami) [inline app/cdap/components/GridTable/components/Breadcrumb/__tests__/Breadcrumb.test.tsx]: remove any commented code

File: app/cdap/components/GridTable/types.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ export interface IParams {
 }
 
 export interface IRecords {
-  [key: string]: string;
+  [key: string]: string | any;
```
Comment (adityadivami) [inline app/cdap/components/GridTable/types.ts]: update any to proper interface

File: app/cdap/components/GridTable/utils.tsx [hunk 6]
```diff
@@ -47,30 +48,32 @@ export const checkFrequentlyOccuredValues = (
...
-    valueOfKey.map((item, index) => {
-      valueOfKey.map((value, valueIndex) => {
-        if (item == value) {
-          mostFrequentItemCount++;
-        }
-        if (mostFrequentItem < mostFrequentItemCount) {
-          mostFrequentItem = mostFrequentItemCount;
-          mostFrequentItemValue = item;
-        }
+  if (gridData?.values && Array.isArray(gridData?.values)) {
```
Comment (adityadivami) [inline app/cdap/components/GridTable/utils.tsx:51]: what if if condition failes, we should handle the return value

File: package.json [hunk 7]
```diff
@@ -36,7 +36,7 @@
     "cypress": "cypress run",
```
Comment (adityadivami) [inline package.json]: do not push package.json please

File: app/cdap/components/GridTable/__tests__/GridTable.test.tsx [hunk 8]
```diff
@@ -16,26 +16,328 @@
...
-  const { getByTestId } = render(
-    <Router history={history}>
-      <Switch>
-        <Route>
-          <GridTable />
-        </Route>
-      </Switch>
-    </Router>
-  );
+  jest.spyOn(rxjs, 'flatMap' as any).mockImplementation((callback: any) => {
```
Comment (adityadivami) [inline app/cdap/components/GridTable/__tests__/GridTable.test.tsx:31]: change any to appropriate interface

File: app/cdap/components/GridTable/__tests__/GridTable.test.tsx (hunk 1 above)
Comment (smritiDivami) [reply app/cdap/components/GridTable/__tests__/GridTable.test.tsx:32]: Done

File: app/cdap/components/GridTable/__tests__/GridTable.test.tsx (hunk 2 above)
Comment (smritiDivami) [reply app/cdap/components/GridTable/__tests__/GridTable.test.tsx]: Done

File: app/cdap/components/GridTable/__tests__/GridTable.test.tsx (hunk 3 above)
Comment (smritiDivami) [reply app/cdap/components/GridTable/__tests__/GridTable.test.tsx]: Checked

File: app/cdap/components/GridTable/utils.tsx (hunk 6 above)
Comment (smritiDivami) [reply app/cdap/components/GridTable/utils.tsx:51]: added return {name:"No Data Found",count:0};

File: package.json (hunk 7 above)
Comment (smritiDivami) [reply package.json]: Done

File: app/cdap/components/GridTable/utils.tsx (hunk 6 above)
Comment (smritiDivami) [reply app/cdap/components/GridTable/utils.tsx:51]: Or shall I add {name:"Nil",count:0}

**Pull request closed without merging**